ai_ta_backend/update_materials.py

```diff
@@ -0,0 +1,96 @@
+import os
+import shutil
+import boto3
+from ai_ta_backend.vector_database import Ingest
+import hashlib
+from ai_ta_backend.aws import upload_data_files_to_s3
+from ai_ta_backend.vector_database import Ingest
+
+
+def generate_checksum(file_path):
+    md5_hash = hashlib.md5()
+    with open(file_path, "rb") as file:
+        # Read and update the hash string value in blocks
+        for byte_block in iter(lambda: file.read(4096), b""):
+            md5_hash.update(byte_block)
+    return md5_hash.hexdigest()
+
+def update_files(source_path: str, course_name: str):

```

Comment:
Oh I'm so excited to have this function!

Reply:
But please update it to also delete the old files if they're being replaced. We have a function for this (it's called in the /delete endpoint).

Reply:
Okay! Right now I delete the newly downloaded file if its checksum matches with any of the existing file and upload the rest (with non-matching checksums). 
I will add an extra condition - if the checksum does not match, but the filename already exists, replace the old file with the new.

Reply:
This is done!